app/services/user_service.rb

```diff
@@ -0,0 +1,58 @@
+module UserService
+  def self.create(params:)
+    new_user = _user_from_params(params)
+
+    if new_user.performer?
+      User.transaction do
+        old_user = User.find_by(current_spot: new_user.current_spot)
+        deactivate_user(buck_id: old_user.buck_id) unless old_user.nil?
+      end
+    end
+
+    new_user.save! if new_user.valid?
+    new_user
+  end
+
+  def self.deactivate_user(buck_id:)
+    user = User.find(buck_id)
+    user.active = false
```

Comment:
This could be shorter if you want:

```ruby
user = User.find(buck_id)
user.update!(active: false, current_spot: nil, original_spot: nil)
user
```